Quest-3/Part-1/main.py

```diff
@@ -0,0 +1,71 @@
+def main(fileName):
+    lines = readFile(fileName)
+
+    schemEngine = [list(line.rstrip("\n")) for line in lines]
+    symbols = set(char for line in schemEngine for char in line if char != '.' and not char.isdigit())
+
+    allNumbers = findNumbers(schemEngine)
+    trueNumbers = findSymbolsAroundNumbers(schemEngine, allNumbers, symbols)
+
+    total = sum(trueNumbers)
+    print(total)
+
+def readFile(name):
+    with open(file=f"Quest-3/{name}", mode="r", encoding="utf-8") as f:
+        return f.readlines()
+
+def findNumbers(matrix):
+    numbers = []
+    visited = set()
+    rows = len(matrix)
+    cols = len(matrix[0])
+
+    def explore(row, col):
+        num = []
+        num.append({"digit": matrix[row][col], "coordinates": [row, col]})
+        # Explore horizontally
+        for c in range(col + 1, cols):
+            if matrix[row][c].isdigit() and (row, c) not in visited:
+                num.append({"digit": matrix[row][c], "coordinates": [row, c]})
+                visited.add((row, c))
+            else:
+                break
+        return num
+
+    for row in range(rows):
+        for col in range(cols):
+            if matrix[row][col].isdigit() and (row, col) not in visited:
```

Comment:
you check for this in `explore` too. maybe your explore function could do more? and return None if there's no number.